src/Fantomas/AstTransformer.fs

```diff
@@ -62,1634 +34,1211 @@ module private Ast =
+                    |> finalContinuation
+                | SynOpenDeclTarget.Type (synType, _range) ->
+                    mkNode SynModuleDecl_OpenType parentRange
+                    :: (visitSynType synType)
+                    |> finalContinuation
+            | SynModuleDecl.Attributes (attrs, range) ->
+                mkNode SynModuleDecl_Attributes range
+                :: (visitSynAttributeLists range attrs)
+                |> finalContinuation
+            | SynModuleDecl.HashDirective (hash, range) ->
+                [ mkNode SynModuleDecl_HashDirective range
+                  visitParsedHashDirective hash ]
+                |> finalContinuation
+            | SynModuleDecl.NamespaceFragment (moduleOrNamespace) ->
+                visitSynModuleOrNamespace moduleOrNamespace
+                |> finalContinuation
+
+        visit ast id
+
+    and visitSynExpr (synExpr: SynExpr) : TriviaNodeAssigner list =
+        let rec visit
+            (synExpr: SynExpr)
+            (finalContinuation: TriviaNodeAssigner list -> TriviaNodeAssigner list)
+            : TriviaNodeAssigner list =
+            match synExpr with
+            | SynExpr.Paren (expr, _, _, range) ->
+                visit
+                    expr
+                    (fun nodes ->
+                        [ mkNode SynExpr_Paren range
+                          yield! nodes ]
+                        |> finalContinuation)
+            | SynExpr.Quote (operator, _, quotedSynExpr, _, range) ->
+                let continuations : ((TriviaNodeAssigner list -> TriviaNodeAssigner list) -> TriviaNodeAssigner list) list =
+                    [ visit operator; visit quotedSynExpr ]
+
+                let finalContinuation (nodes: TriviaNodeAssigner list list) : TriviaNodeAssigner list =
+                    [ mkNode SynExpr_Quote range
+                      yield! (List.collect id nodes) ]
+                    |> finalContinuation
+
+                Continuation.sequence continuations finalContinuation
+            | SynExpr.Const (constant, range) ->
+                [ mkNode SynExpr_Const range
+                  visitSynConst range constant ]
+                |> finalContinuation
+            | SynExpr.Typed (expr, typeName, _) ->
+                visit
+                    expr
+                    (fun nodes ->
+                        //                        { Type = SynExpr_Typed
+//                           Range = r range
+//                           Properties = Map.empty
+//                            }
+                        nodes @ visitSynType typeName |> finalContinuation)
+            | SynExpr.Tuple (_, exprs, _, range) ->
+                let continuations : ((TriviaNodeAssigner list -> TriviaNodeAssigner list) -> TriviaNodeAssigner list) list =
+                    exprs |> List.map visit
+
+                let finalContinuation (nodes: TriviaNodeAssigner list list) : TriviaNodeAssigner list =
+                    [ mkNode SynExpr_Tuple range
```

Comment:
Would `mkNode SynExpr_Tuple range :: (List.collect id nodes)` be better? I seem to switch styles here and there.

Reply:
Pff, I really don't mind. Personally I prefer to be explicit if I use the `yield` keyword - I'm not a fan of implicit yields, because if you omit the yields then you have the chance in principle to get confused between `[if true then yield 3 else yield 4]` and `[yield (if true then 3 else 4)]`.

I suspect `mkNode SynExpr_Tuple range :: (List.collect id nodes)` might be more efficient - it would certainly rely less on the compiler's magic ability to optimise. It's not obvious to me that the computation expression avoids copying whatever is `yield!`'ed.